Code.js

```diff
@@ -0,0 +1,193 @@
+var FREQUENCY = {
+  DAILY: 1,
+  WEEKLY: 2,
+  PAUSED: 3,
+};
+
+var SHEET_DATA = "Data";
+
+//return all participants from the Sheet_Data
+function getParticipants() {
+  var range = "A2:D";
+  var rangeValues = SpreadsheetApp.getActiveSpreadsheet()
+    .getSheetByName(SHEET_DATA)
+    .getRange(range)
+    .getValues();
+  var participants = [];
+
+  rangeValues.forEach((row) => {
+    if (row[1] != "") {
+      participants.push(new Participant(row));
+    }
+  });
+
+  return participants;
+}
+
+function getTeams(participants) {
+  var teams = [];
+
+  participants.forEach((p) => {
+    var t = p.team;
+    var n = p.name;
+    if (!(t in teams)) {
+      teams[t] = [];
+    }
+
+    teams[t].push(p);
+  });
+
+  return teams;
+}
+
+function shuffleArray(array) {
+  for (let i = array.length - 1; i > 0; i--) {
+    const j = Math.floor(Math.random() * (i + 1));
+    [array[i], array[j]] = [array[j], array[i]];
+  }
+}
+
+//look up user id from the slack user list by email 
+function getSlackUserID (participant, slack) {
+  var id = slack.getUserID(participant.email);
+  if (id == '') { id = participant.toString(); }
```

Comment:
we tend to use === and !== for comparisons in JS, but I don't think it matters too much in your case as everything should be strings

Reply:
that's a very good point - i forgot about that!